packages/ui/src/components/theme-selector-v2/theme-dialog.tsx

```diff
@@ -0,0 +1,181 @@
+  onOpenChange,
+  onChange,
+  onSave,
+  onCancel,
+  children
+}) => {
+  const [mode, setMode] = useState<Mode>(Mode.Dark)
+  const [contrast, setContrast] = useState<Contrast>(Contrast.Default)
+  const [colorAdjustment, setColorAdjustment] = useState<ColorAdjustment>(ColorAdjustment.Default)
+  const [accentColor, setAccentColor] = useState<AccentColor>(AccentColor.Blue)
+
+  useEffect(() => {
+    if (theme) {
+      setMode(theme.mode)
+      setContrast(theme.contrast)
+      setColorAdjustment(theme.colorAdjustment)
+      setAccentColor(theme.accentColor)
+    } else if (defaultTheme) {
+      setMode(defaultTheme.mode)
+      setContrast(defaultTheme.contrast)
+      setColorAdjustment(defaultTheme.colorAdjustment)
+      setAccentColor(defaultTheme.accentColor)
+    }
+  }, [defaultTheme, theme])
+
+  return (
+    <Dialog.Root open={open} onOpenChange={onOpenChange}>
+      <Dialog.Trigger asChild>{children}</Dialog.Trigger>
+      <Dialog.Content className="w-[550px]">
+        <Dialog.Title>Appearance Settings</Dialog.Title>
+        {/* Mode */}
+        <div>
+          <div className="flex flex-col gap-1">
+            <Text className="text-md font-medium">Mode</Text>
+            <Text className="text-xs text-foreground-3">Select or customize your UI theme.</Text>
+          </div>
+          <div className="mt-4 flex gap-4">
+            {Object.values(Mode).map(item => (
+              <label key={item} className="flex flex-col items-start cursor-pointer gap-2">
+                <div
+                  className={cn(
+                    'h-[131px] w-[225px] rounded border p-2',
+                    mode === item ? 'border-white' : 'border-gray-600',
+                    item === Mode.Light ? 'bg-gray-300' : 'bg-black'
+                  )}
+                />
+                <div className="flex items-center gap-2">
+                  <input
+                    type="radio"
+                    name="theme"
+                    value={item}
+                    checked={mode === item}
+                    onChange={() => {
+                      setMode(item)
+                      onChange({ mode: item, contrast, colorAdjustment, accentColor })
+                    }}
+                    className="hidden"
+                  />
+                  <div className="h-4 w-4 rounded-full border border-gray-600 flex items-center justify-center">
+                    {mode === item && <div className="h-2 w-2 rounded-full bg-white" />}
+                  </div>
+                  {item}
+                </div>
+              </label>
+            ))}
+          </div>
+        </div>
+
+        <Separator className="h-px bg-gray-800" />
+
+        {/* Contrast */}
+        <div>
+          <div className="flex flex-col gap-1">
+            <Text className="text-md font-medium">Contrast</Text>
+            <Text className="text-xs text-foreground-3">
+              High contrast improves readability, Dimmer mode reduces glare.
```

Comment:
we should probably just call it `Dim` mode instead of `Dimmer`

Reply:
Done in https://github.com/harness/canary/pull/921.